Documentation.md

```diff
@@ -164,590 +174,606 @@ and made available to your JS client code.
 
+* `agentStateARN`: The agent's current state ARN.
 * `name`: The name of the agent's current availability state.
+* `startTimestamp`: A `Date` object that indicates when the state was set.
 * `type`: The agent's current availability state type, as per the `AgentStateType` enumeration.
-* `duration`: A relative local state duration.  To get the actual duration of the state relative
-  to the current time, use `agent.getStateDuration()`.
 
-### `agent.getStateDuration()`
-```
+### `agent.getStateDuration()` / `agent.getStatusDuration()`
+```js
 var millis = agent.getStateDuration();
 ```
-Get the duration of the agent's state in milliseconds relative to local time.  This takes into
+Get the duration of the agent's state in milliseconds relative to local time. This takes into
 account time skew between the JS client and the Amazon Connect service.
 
 ### `agent.getPermissions()`
+```js
+var permissions = agent.getPermissions(); // e.g. ["outboundCall"]
 ```
-var permissions = agent.getPermissions();
-```
-Mostly for internal purposes only.  Contains strings which indicates actions that the agent can
+Mostly for internal purposes only. Contains strings which indicates actions that the agent can
 take in the CCP.
 
 ### `agent.getContacts()`
-```
+```js
 var contacts = agent.getContacts(contactTypeFilter);
 ```
 Gets a list of `Contact` API objects for each of the agent's current contacts.
 
-* `contactTypeFilter`: Optional.  If provided, only contacts of the given type are returned.
+* `contactTypeFilter`: Optional. If provided, only contacts of the given `ContactType` enum are returned.
 
 ### `agent.getConfiguration()`
-```
+```js
 var config = agent.getConfiguration();
 ```
-Gets the full `AgentConfiguration` object for the agent.  This object contains the following fields:
+Gets the full `AgentConfiguration` object for the agent. This object contains the following fields:
 
-* `name`: The agent's user friendly display name.
-* `softphoneEnabled`: Indicates whether the agent's phone calls should route to the agent's
-  browser-based softphone or the telephone number configured as the agent's extension.
-* `extension`: Indicates the phone number that should be dialed to connect the agent to their
-  inbound or outbound calls when softphone is not enabled.
-* `routingProfile`: Describes the agent's current routing profile and list of
-  queues therein. See `agent.getRoutingProfile()` for more info.
+* `agentStates`: See `agent.getAgentStates()` for more info.
+* `dialableCountries`: See `agent.getDialableCountries()` for more info.
+* `extension`: See `agent.getExtension()` for more info.
+* `name`: See `agent.getName()` for more info.
+* `permissions`: See `agent.getPermissions()` for more info.
+* `routingProfile`: See `agent.getRoutingProfile()` for more info.
+* `softphoneAutoAccept`: Indicates whether auto accept soft phone calls is enabled.
+* `softphoneEnabled`: See `agent.isSoftphoneEnabled()` for more info.
```

Comment:
Found out some fields were missing, and since all of them were obtainable through individual `.get*()` methods, referred to their documentation instead.